### PR TITLE
fix: Gradle publish script

### PR DESCRIPTION
**Pull request by hamada147**
# Overview

Re-wrote the Gradle publish script in another manner which should now works.

Fixes #<issue number>

### My PR contains...
* [ ] No code changes (changes to documentation, CI, metadata, etc.)
* [ ] Bug fixes (non-breaking change which fixes an issue)
* [ ] Improvements (misc. changes to existing features)
* [ ] Features (non-breaking change which adds functionality)

### My changes...
* [ ] are breaking changes
* [ ] are not breaking changes
* [ ] If yes to above: I have updated the documentation accordingly

### Documentation
* [ ] My changes do not require a change to the project documentation
* [ ] My changes require a change to the project documentation
* [ ] If yes to above: I have updated the documentation accordingly

### Tests
* [ ] My changes can not or do not need to be tested
* [ ] My changes can and should be tested by unit and/or integration tests
* [ ] If yes to above: I have added tests to cover my changes
* [ ] If yes to above: I have taken care to cover edge cases in my tests

### CODE REVIEW

**Comment by atala-dev** [comment]
### Code Coverage
|Total Project Coverage|91.66%|:white_check_mark:|
|:-|:-:|:-:|

File: apollo/build.gradle.kts [hunk 1]
```diff
@@ -390,20 +392,30 @@ afterEvaluate {
     }
     // Disable publish of targets
     if (os.isMacOsX) {
-        tasks.named("publishIosX64PublicationToSonatypeRepository") {
-            this.enabled = false
+        if (tasks.findByName("publishIosX64PublicationToSonatypeRepository") != null) {
+            tasks.named("publishIosX64PublicationToSonatypeRepository") {
+                this.enabled = false
+            }
```
Comment (goncalo-frade-iohk) [inline apollo/build.gradle.kts:398]: Wonder with this new if's you don't need the `if(os.isMacOsX)`

**Review by goncalo-frade-iohk — APPROVED** [review]

File: apollo/build.gradle.kts (hunk 1 above)
Comment (hamada147) [reply apollo/build.gradle.kts:398]: We need the if condition because we are also running the project on a `Linux` environment in which these Gradle tasks do not exist.
